config/v1/types.go

```diff
@@ -5,6 +5,84 @@ import (
+	// background per minute. The default value is 60. Set to -1 for unlimited.
+	MaxScheduledImageImportsPerMinute int32 `json:"maxScheduledImageImportsPerMinute"`
+	// AllowedRegistriesForImport limits the docker registries that normal users may import
+	// images from. Set this list to the registries that you trust to contain valid Docker
+	// images and that you want applications to be able to import from. Users with
+	// permission to create Images or ImageStreamMappings via the API are not affected by
+	// this policy - typically only administrators or system integrations will have those
+	// permissions.
+	AllowedRegistriesForImport []RegistryLocation `json:"allowedRegistriesForImport,omitempty"`
+	// InternalRegistryHostname sets the hostname for the default internal image
+	// registry. The value must be in "hostname[:port]" format.
+	// For backward compatibility, users can still use OPENSHIFT_DEFAULT_REGISTRY
+	// environment variable but this setting overrides the environment variable.
+	InternalRegistryHostname string `json:"internalRegistryHostname,omitempty"`
+	// ExternalRegistryHostname sets the hostname for the default external image
+	// registry. The external hostname should be set only when the image registry
+	// is exposed externally. The value is used in 'publicDockerImageRepository'
+	// field in ImageStreams. The value must be in "hostname[:port]" format.
+	ExternalRegistryHostname string `json:"externalRegistryHostname,omitempty"`
+	// AdditionalTrustedCA is a path to a pem bundle file containing additional CAs that
+	// should be trusted during imagestream import.
+	AdditionalTrustedCA ConfigMapReference `json:"additionalTrustedCA,omitempty"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type ImageConfigList struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []ImageConfig `json:"items"`
+}
+
+// RegistryLocation contains a location of the registry specified by the registry domain
+// name. The domain name might include wildcards, like '*' or '??'.
+type RegistryLocation struct {
+	// DomainName specifies a domain name for the registry
+	// In case the registry use non-standard (80 or 443) port, the port should be included
+	// in the domain name as well.
+	DomainName string `json:"domainName"`
+	// Insecure indicates whether the registry is secure (https) or insecure (http)
+	// By default (if not specified) the registry is assumed as secure.
+	Insecure bool `json:"insecure,omitempty"`
+}
+
+// ConfigMapReference references the location of a configmap.
+type ConfigMapReference struct {
+	Namespace string `json:"namespace"`
+	Name      string `json:"name"`
+	// FileName allows pointing to a specific file inside of the configmap.  This is useful for logical file references.
+	FileName string `json:"filename,omitempty"`
```

Comment:
why isn't this just a key within the configmap?

in what sense does a configmap even contain "files"?  (doesn't it just contain keys+values where the values can be mounted as files?)

Reply:
> in what sense does a configmap even contain "files"? (doesn't it just contain keys+values where the values can be mounted as files?)

I think most people who use them think of the keys as filenames since mounting a configmap mounts each key as a file.

Reply:
but there's no obligation here that the configmap be mounted.  At the point in time at which you are declaring the reference, you are declaring a reference to a configmap key. 

The consumer of this configmapreference could choose to mount that key into a pod, or read it directly.  I think it's incorrect to claim it's a filename here unless the only valid way to use this configmapreference is to mount that specific key as that specific filename.

Reply:
> but there's no obligation here that the configmap be mounted. At the point in time at which you are declaring the reference, you are declaring a reference to a configmap key.
> 
> The consumer of this configmapreference could choose to mount that key into a pod, or read it directly. I think it's incorrect to claim it's a filename here unless the only valid way to use this configmapreference is to mount that specific key as that specific filename.

No issue with renaming it.